spinnaker/spinnaker_system/google_front50_test.py

```diff
@@ -158,32 +179,46 @@ def update_app(self):
       elif isinstance(value, basestring):
         spec[name] = 'NEW_' + value
     payload = self.agent.make_json_payload_from_object(spec)
-    expect = dict(spec)
+    expectUpdate = dict(spec)
 
     # The actual update is determined by front50.
     # The createTs we gave ignored.
     # As before, the name is upper-cased.
-    del expect['updateTs']
-    expect['createTs'] = self.initial_app_spec['createTs']
-    expect['name'] = self.initial_app_spec['name'].upper()
+    del expectUpdate['updateTs']
+    expectUpdate['createTs'] = self.initial_app_spec['createTs']
+    expectUpdate['name'] = self.initial_app_spec['name'].upper()
+    self.app_history.insert(0, expectUpdate)
 
     # TODO(ewiseblatt) 20160524:
     # Add clauses that observe Front50 to verify the history method works
     # and that the get method is the current version.
     num_versions = 2 if self.versioning_enabled else 1
-    builder = gcp.GcpStorageContractBuilder(self.gcs_observer)
-    (builder.new_clause_builder('Google Cloud Storage Contains File')
+    gcs_builder = gcp.GcpStorageContractBuilder(self.gcs_observer)
+    (gcs_builder.new_clause_builder('Google Cloud Storage Contains File')
      .list_bucket(self.BUCKET,
                   '/'.join([self.BASE_PATH, 'applications', self.TEST_APP]),
                   with_versions=True)
      .contains_path_value('name', self.TEST_APP,
                           min=num_versions, max=num_versions))
-    (builder.new_clause_builder('Updated File Content')
+    (gcs_builder.new_clause_builder('Updated File Content')
      .retrieve_content(self.BUCKET,
                        '/'.join([self.BASE_PATH, 'applications', self.TEST_APP,
                                  'specification.json']),
                        transform=json.JSONDecoder().decode)
-     .contains_path_value('', expect))
+     .contains_path_value('', expectUpdate))
+
+    for clause in gcs_builder.build().clauses:
+      contract.add_clause(clause)
+
+    f50_builder = st.http_observer.HttpContractBuilder(self.agent)
+    (f50_builder.new_clause_builder('History Records Changes')
+     .get_url_path('/default/applications/{app}/history'
+                   .format(app=self.TEST_APP))
+     .contains_path_value('[0]', self.app_history[0])
+     .contains_path_value('[1]', self.app_history[1]))
```

Comment:
The historic entities were remembered properly.
Path '[N]' means the Nth element in a list. So this is checking the list reflects the value and ordering.